### ISSUE TITLE
"user not logged in" - no database functionality for custom book

### ISSUE THREAD

**Issue by korin-richmond** [RunestoneInteractive/RunestoneServer#1323]
tl;dr - database functionality (e.g. tracking a logged-in user's activity) works when a user is using "thinkcspy", but not when they change course to use the custom book I created ("cpslp").

More detail...
A couple of years ago I made a custom course from scratch and installed RunestoneServer to locally serve it to my class.  That worked great for a couple of years.  Students could log in and all their activity would be saved to the database.

A couple of days ago, I did a fresh install of everything RunestoneServer (incl. web2py, RunestoneComponents, etc...) to bring everything up to date.  I also completely reset and reinitialised my postgres database.

I followed the instructions in the README.rst in this repository.  I then installed the "thinkcspy" and my "cpslp" books into the "books" subdirectory, built and deployed them, then created a user account for myself and registered for the two courses.

Alas, when I log in all the database functionality works completely fine with "thinkcspy", but when I navigate to "change course" and select "cpslp", that all stops.  All pages have "user not logged in" at the bottom - even though I *am* still logged in (if I switch course back to "thinkcspy", it's all still fine). 

I've checked and re-checked the pavement.py config files - I can't see any difference between the one for "thinkcspy" and my "cpslp" (e.g. dynamic_pages, login_required, use_services, all set to true).

There must be some difference, but I've been looking everywhere and haven't found what it is.  Is it possible there's something hardcoded somewhere else in runestone (outside my book directory) which means thinkcspy works, while the custom book doesn't?

I'd be hugely, hugely grateful for any pointers how to diagnose and fix this, or indeed suggestions for things I may just have missed!

Thanks!

Korin

**What Course are you in**
thinkcspy, and my own completely custom course written from scratch.

**Comment by bnmnetp** [comment]
Korin,

We made some big changes in the architecture of Runestone this summer.  The problem you are having is related to "dynamic pages". We moved away from building a separate static textbook for each course and now the server fills in the course specific stuff when the page is served.

It sounds like you are up to speed on the changes to the conf.py and pavement.py files, but the logged out message tells me that you are not getting the right templates.

Your course should be serving from server/runestone/books/published -- if your book is serving from /runestone/static/....  remove the folder for your course from static

**Comment by korin-richmond** [comment]
Your analysis was spot on - my book wasn't using the right templates, as I had overlooked that necessary update in the conf.py file. 

So now I have set templates_path and html_theme_path to:
templates_path = [pkg_resources.resource_filename('runestone', 'common/project_template/_templates')]
html_theme_path = [pkg_resources.resource_filename('runestone', 'common/project_template/_templates/plugin_layouts')]

(these were previously just '_templates' and '_templates/plugin_layouts' respectively)

And now the issue is resolved and everything works brilliantly again - thanks so much for your speedy insightful comment!

**Comment by bnmnetp** [comment]
Great!
